app/imports/ui/pages/JobListing.jsx

```diff
@@ -37,17 +27,10 @@ const JobListing = () => {
         </Col>
         <Col>
           <Row>
-            { sampleCompanies.map(({ image, title, company, description, link }, index) => (
-              <Col xs={12} md={4} key={index}>
-                <Card className="justify-content-center" id="listing-card">
-                  <Card.Img id="listing-card-image" variant="top" src={image} />
-                  <Card.Title id="listing-card-title">{title}</Card.Title>
-                  <Card.Text id="listing-card-text">Company: {company}</Card.Text>
-                  <Card.Text id="listing-card-text">{description}</Card.Text>
-                  <Card.Link id="listing-card-link" to={link}><Button className="visit-button">More Info</Button></Card.Link>
-                </Card>
-              </Col>
-            )) }
+            { ready ? (listings.map((listing, index) => (
+              <Listing listing={listing} key={index} />
+            )))
+              : (<LoadingSpinner />) }
```

Comment:
This should be:
```jsx
{ ready ? (listings.map((listing) => (
  <Listing listing={listing} key={listing._id} />
))) : <LoadingSpinner /> }
```

The way you are doing it, id is just the index of the array, which is not an identifier of the job listing document.